backend/api/Services/MissionSchedulingService.cs

```diff
@@ -102,6 +102,32 @@ public async Task StartNextMissionRunIfSystemIsAvailable(string robotId)
                 return;
             }
 
+            if ((!robot.IsRobotPressureHighEnoughToStartMission() || !robot.IsRobotBatteryLevelHighEnoughToStartMissions()) && !missionRun.IsReturnHomeMission())
```

Comment:
Could we maybe move this block to another function? It's subjective so it's up to you, but the current function is quite big.